src/cluster.c

```diff
@@ -126,6 +132,85 @@ static ConnectionType *connTypeOfCluster() {
 
     return connectionTypeTcp();
 }
+/* Cluster shards hash table, mapping shard id to list of nodes */
+dictType clusterSdsToListType = {
+        dictSdsHash,                /* hash function */
+        NULL,                       /* key dup */
+        NULL,                       /* val dup */
+        dictSdsKeyCompare,          /* key compare */
+        dictSdsDestructor,          /* key destructor */
+        dictListDestructor,         /* val destructor */
+        NULL                        /* allow to expand */
+};
+
+/* Aux fields are introduced in Redis 7.2 to support the persistence
+ * of various important node properties, such as shard id, in nodes.conf.
+ * Aux fields take an explicit format of name=value pairs and have no
+ * intrinsic order among them. Aux fields are always grouped together
+ * at the end of the second column of each row after the node's IP
+ * address/port/cluster_port and the optional hostname. Aux fields
+ * are separated by ','. */
+
+/* Aux field setter function prototype
+ * return 1 when the update is successful; 0 otherwise */
```

Comment:
@madolson any suggestion on the return value from the setter callback? would C_OK/C_ERR be more preferable? I am essentially using "boolean" here.

Reply:
IMHO boolean is good if the name of the function hints that it's a boolean, such as `isValidAuxChar`, whilst for a setter I think OK/ERR is better (or void if it can't fail).

Reply:
make sense. yeah this setter can definitely fail (for malformed values).